util/util.go

```diff
@@ -12,22 +13,44 @@ import (
 	"regexp"
 	"strings"
 	"time"
+
+	"github.com/nknorg/tuna"
 )
 
-func GetFreePort() (int, error) {
-	addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
-	if err != nil {
-		return 0, err
-	}
+// Get free port start from parameter port
+// If paramenter port is 0, start from system returned free port
+// The returned port is free in TCP and UDP
+func GetFreePort(port int) (int, error) {
+	for i := 0; i < 100; i++ {
```

Comment:
retry without sleep is not likely to make a difference. We should add sleep (but also reduce the number of retries so it won't take forever)

Reply:
Here each retry will increase port. Such as we first retry 1080, if it is busy, then try 1081, 1082, ... until we get a free port. Is it suitable?

Reply:
Where is the code to increase port? And since you don't need a fixed port, why not a random dynamic port?

Reply:
when UDP listen fails, the port will be increased to try the next port. if argument `port`=0, it will try a random dynamic port. This modification expands the use cases of this function: 1. get a free port from system random port; 2 get a specific port or subsequent port, such as 1080, 1081, ...

`
if err != nil {
			l.Close()
			port++
			continue
}
`